src/Algebra/Properties/CommutativeSemigroup.agda

```diff
@@ -0,0 +1,121 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some theory for commutative semigroup
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (Semigroup)
+open import Algebra.FunctionProperties using (Commutative)
+
+module Algebra.Properties.CommutativeSemigroup {ℓ ℓ=} (H : Semigroup ℓ ℓ=)
+                                     (open Semigroup H using (_≈_; _∙_))
+                                     (comm : Commutative _≈_ _∙_)
+                                     where
+
+import      Relation.Binary.Properties.Setoid as OfSetoid
+open import Function using (_$_)
+import      Relation.Binary.EqReasoning as EqR
+
+open Semigroup H using (refl; sym; trans; assoc; ∙-congˡ; ∙-congʳ; setoid)
+open OfSetoid setoid public
+open EqR setoid
+
+-- Permutation laws for _∙_ for three factors.
+
+------------------------------------------------------------------------------
+-- Partitions (1,1).
+-- There are five nontrivial permutations.
+------------------------------------------------------------------------------
+
+x∙yz≈y∙xz :  ∀ x y z → x ∙ (y ∙ z) ≈ y ∙ (x ∙ z)    -- x-y
+x∙yz≈y∙xz x y z = begin
+  x ∙ (y ∙ z)    ≈⟨ sym (assoc x y z) ⟩
+  (x ∙ y) ∙ z    ≈⟨ ∙-congʳ (comm x y) ⟩
+  (y ∙ x) ∙ z    ≈⟨ assoc y x z ⟩
+  y ∙ (x ∙ z)    ∎
+
+x∙yz≈z∙yx :  ∀ x y z → x ∙ (y ∙ z) ≈ z ∙ (y ∙ x)    -- x-z
+x∙yz≈z∙yx x y z = begin
+  x ∙ (y ∙ z)    ≈⟨ ∙-congˡ (comm y z) ⟩
+  x ∙ (z ∙ y)    ≈⟨ x∙yz≈y∙xz x z y ⟩
+  z ∙ (x ∙ y)    ≈⟨ ∙-congˡ (comm x y) ⟩
+  z ∙ (y ∙ x)    ∎
+
+x∙yz≈x∙zy :  ∀ x y z → x ∙ (y ∙ z) ≈ x ∙ (z ∙ y)    -- y-z
+x∙yz≈x∙zy _ y z =  ∙-congˡ (comm y z)
+
+x∙yz≈y∙zx :  ∀ x y z → x ∙ (y ∙ z) ≈ y ∙ (z ∙ x)    -- cycle
+x∙yz≈y∙zx x y z = begin
+  x ∙ (y ∙ z)   ≈⟨ comm x _ ⟩
+  (y ∙ z) ∙ x   ≈⟨ assoc y z x ⟩
+  y ∙ (z ∙ x)   ∎
+
+x∙yz≈z∙xy :  ∀ x y z → x ∙ (y ∙ z) ≈ z ∙ (x ∙ y)
+x∙yz≈z∙xy x y z = begin
+  x ∙ (y ∙ z)   ≈⟨ sym (assoc x y z) ⟩
+  (x ∙ y) ∙ z   ≈⟨ comm _ z ⟩
+  z ∙ (x ∙ y)   ∎
+
+------------------------------------------------------------------------------
+-- Partitions (1,2).
+-- These permutation laws are proved by composing the proofs for
+-- partitions (1,1) with  \p → trans p (sym (assoc _ _ _)).
+------------------------------------------------------------------------------
+x∙yz≈xy∙z =  \x y z → sym (assoc x y z)
```

Comment:
This belongs in `Algebra.Properties.Semigroup` as it does not rely on commutativity. You should also give it a type.

Reply:
Thank you.
Added  Algebra/Properties/Semigroup,  and ``x∙yz≈xy∙z`` moved there,
and the type is given.